src/lib/services/LocalCatalog.ts

```diff
@@ -130,7 +130,7 @@ export default class LocalCatalog implements RuleCatalog {
 			await this.rebuildCatalogIfNecessary();
 			this.catalog = await this.readCatalogJson();
 		}
-		return Promise.resolve(this.catalog);
```

Comment:
I came across this while looking at code. My guess is it was previously added when the method wasn't async. But now the method is async and it just confuses the reader.

Reply:
Good catch. Do you want to do this in the `rebuildCatalog()` method too?